mac/Keyman4MacIM/Keyman4MacIM/KMConfiguration/KMConfigurationWindowController.m

```diff
@@ -409,19 +409,22 @@ - (void)helpAction:(id)sender {
 }
 
 - (void)removeAction:(id)sender {
-  NSButton *removeButton = (NSButton *)sender;
-  NSDictionary *info = [self.tableContents objectAtIndex:removeButton.tag];
+  NSButton *deleteButton = (NSButton *)sender;
+  NSDictionary *info = [self.tableContents objectAtIndex:deleteButton.tag];
   NSString *deleteKeyboardMessage = NSLocalizedString(@"message-confirm-delete-keyboard", nil);
   
   if ([info objectForKey:@"HeaderTitle"] != nil)
     [self.deleteAlertView setMessageText:[NSString localizedStringWithFormat:deleteKeyboardMessage, [info objectForKey:@"HeaderTitle"]]];
   else
     [self.deleteAlertView setMessageText:[NSString localizedStringWithFormat:deleteKeyboardMessage, [info objectForKey:kKMKeyboardNameKey]]];
   
-  [self.deleteAlertView beginSheetModalForWindow:self.window
-                                   modalDelegate:self
-                                  didEndSelector:@selector(alertDidEnd:returnCode:contextInfo:)
-                                     contextInfo:(__bridge void *)([NSNumber numberWithInteger:removeButton.tag])];
+  [self.deleteAlertView beginSheetModalForWindow:self.window completionHandler:^(NSModalResponse returnCode)  {
+    if (returnCode == NSAlertFirstButtonReturn) {
+      os_log_debug([KMLogs uiLog], "confirm delete keyboard alert dismissed");
+      [self deleteFileAtIndex:[NSNumber numberWithInteger:deleteButton.tag]];
+      self.deleteAlertView = nil;
+    }
+  }];
```

Comment:
Is this unrelated?

Reply:
Not related to keyboard installation, but for this fix I replaced all the deprecated calls to beginSheetModalForWindow with the newer version that behaves slightly differently. The older API was deprecated in macOS 10.10 and we now require 10.13.